### ISSUE TITLE
Circuits with delay cannot be submitted to SamplerV2

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-ibm-runtime#1613]
**Describe the bug**
```python

qc = QuantumCircuit(127)
qc.delay(1,0)
qc.measure_all()

sampler.run(qc, shots=int(1e4))
```

File [~/miniforge3/envs/qiskit/lib/python3.12/site-packages/qiskit/primitives/containers/sampler_pub.py:125](http://localhost:8888/lab/tree/Desktop/miniforge3/envs/qiskit/lib/python3.12/site-packages/qiskit/primitives/containers/sampler_pub.py#line=124), in SamplerPub.coerce(cls, pub, shots)
    123     values = pub[1]
    124     if not isinstance(values, (BindingsArray, Mapping)):
--> 125         values = {tuple(circuit.parameters): values}
    126     parameter_values = BindingsArray.coerce(values)
    127 else:

AttributeError: 'Delay' object has no attribute 'parameters'

**Steps to reproduce**

**Expected behavior**

**Suggested solutions**

**Additional Information**

- **qiskit-ibm-runtime version**: '0.22.0'
- **Python version**:
- **Operating system**:

**Comment by nonhermitian** [comment]
Actually this is a side-effect of seemingly not being able to send a circuit itself to the sampler.run, but rather it must be in list form.  If this is indeed a requirement, a better warning should be produced.  `backend.run` used to take a single circuit without making a len=1 list

**Comment by ihincks** [comment]
The behavior is indeed due to the interface not accepting a bare circuit in the first argument of `run`, which is intended. [This PR](https://github.com/Qiskit/qiskit/pull/12031) introduces a better error for this case, and will become available when qiskit does its 1.0.3 or 1.1 release.

I'll close because I think this gets to the bottom of your issue, but feel free to reopen if I've missed something.
